es5-shim.js

```diff
@@ -1980,9 +1980,15 @@
         parseInt = (function (origParseInt) {
             var hexRegex = /^[\-+]?0[xX]/;
             return function parseInt(str, radix) {
-                var string = trim(str);
-                var defaultedRadix = $Number(radix) || (hexRegex.test(string) ? 16 : 10);
-                return origParseInt(string, defaultedRadix);
```

Comment:
Also, `parseInt('undefined')` should also be `NaN`, so since `parseInt` changes all input to strings, checking the `typeof` isn't very helpful.